File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> st(begin(nums), end(nums));
        int mx = 0;
        for (auto num : nums) {
            int cur = 0;
            if (st.find(num-1) == st.end()) {
                while (st.find(num) != st.end()) {
                    cur++;
                    num++;
                }
                
            }
            mx = max(mx, cur);
        }
        return mx;
    }
};

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        track = set()
        
        for n in nums:
            track.add(n)
        
        mxLen = 1
        cur = 1
        for n in nums:
            if n-1 in track:
                continue
            tmp = n+1
            while tmp in track:
                tmp+=1
                cur+=1
            mxLen = max(mxLen, cur)
            cur = 1
        return mxLen